### PR TITLE
prefixing all options with micm_

### LINKED ISSUE #412: Prefix all cmake build options with MICM_
To prevent conflicts with other packages and make it easy to identify options specific to micm, prefix all options with `MICM_`

**Acceptance criteria**

- All options are prefixed with `MICM_` or something similar.
- All tests and documentation use the new options

### PR DESCRIPTION
**Pull request by K20shores**
To prevent name conflicts with other packages and to make it explicit when using micm from other packages, all micm cmake options are prefixed with `MICM_`

closes #412

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -9,49 +9,51 @@ add_library(musica::micm ALIAS micm)
...
-    message(FATAL_ERROR "GPU_TYPE is not set or is empty. Please provide a GPU type.")
+if(MICM_ENABLE_OPENACC OR MICM_ENABLE_CUDA)
+  if(NOT MICM_GPU_TYPE)
+    message(FATAL_ERROR "MICM_GPU_TYPE is not set or is empty. Please provide a GPU type.")
   endif()
 
   set(GPU_ARCH "")
-  if (GPU_TYPE STREQUAL a100)
+  if (MICM_GPU_TYPE STREQUAL a100)
     set(GPU_ARCH "80")
```
Comment (mwaxmonsky) [inline src/CMakeLists.txt:44]: We should also set `GPU_ARCH` to `MICM_GPU_ARCH` and then I can re-arrange how it's used when I clean up the cuda project cmake to use the newer `project(...)` syntax.

**Review by mwaxmonsky — CHANGES_REQUESTED** [review]
Great work @K20shores!  Just one little update and should be good to go!  Thanks!

**Review by sjsprecious — APPROVED** [review]
Thanks @K20shores . I tested the new options for a GPU build on Derecho and all the tests passed.

**Review by mattldawson — APPROVED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (mattldawson) [reply src/CMakeLists.txt:44]: If this would break the current scripts, I would suggest holding off until you submit the PR with the cleaned-up CUDA build

Comment (K20shores) [reply src/CMakeLists.txt]: @mwaxmonsky done

**Review by mwaxmonsky — APPROVED** [review]
Looks great!

**Pull request merged**